src/edu/washington/escience/myria/io/ByteSink.java

```diff
@@ -0,0 +1,24 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class ByteSink implements DataSink {
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  final ByteArrayOutputStream writerOutput;
```

Comment:
If we materialize this in the constructor (which I don't think we should do), then it should be marked `transient`.